core/src/main/java/io/undertow/protocols/ssl/ALPNHackSSLEngine.java

```diff
@@ -160,7 +160,7 @@ public SSLEngineResult wrap(ByteBuffer[] byteBuffers, int i, int i1, ByteBuffer
                 byte[] data = new byte[newBuf.remaining()];
                 newBuf.get(data);
                 byte[] newData = ALPNHackClientHelloExplorer.rewriteClientHello(data, applicationProtocols);
-                if(newData != null) {
+                if(newData != null && ALPNHackClientByteArrayOutputStream != null) {
```

Comment:
This might not be necessary as those(ssl static methods) are going to be addressed in UNDERTOW-1932

Reply:
Thanks @baranowb for reviewing this. I'll check UNDERTOW-1932 and remove `ALPNHackClientByteArrayOutputStream != null`

Regarding `onOpen != null` with same for onClose, onError
In Line [AnnotatedEndpointFactory#L98](https://github.com/undertow-io/undertow/blob/2.2.9.Final/websockets-jsr/src/main/java/io/undertow/websockets/jsr/annotated/AnnotatedEndpointFactory.java#L98) you can see that in the first run a OnOpen is added to the collection. Line [AnnotatedEndpointFactory#L91](https://github.com/undertow-io/undertow/blob/2.2.9.Final/websockets-jsr/src/main/java/io/undertow/websockets/jsr/annotated/AnnotatedEndpointFactory.java#L91) prevents entering the condition in the first run, but allows it, if L98 has been reached before. Due to that a null pointer will not happen.

Reply:
Yeah, I know, its just irks me, but you are correct.